lib/Core/Executor.cpp

```diff
@@ -905,14 +905,21 @@ Executor::fork(ExecutionState &current, ref<Expr> condition, bool isInternal) {
       }
     }
 
+    // We don't add a constraint explicitly yet
```

Comment:
Yes, initially it's feasible according to both outcomes (the negated state is added to the end of the queue).
But if entered once, only the true branch is possible:
1. `x` is initialized with `1`
2. `(x < y)` must be true if the loop is entered.
3. `(x<y/x)` will be always true and `x *= x` is `1` again.

So we have the same condition as before except that PC is already `(x < y)`.

As no constraint is added to the initial true state as it enters, it won't be removed from the first place, staying there forever. And the negated state can't be selected